machinev2/machine/internal/engine/engine.go

```diff
@@ -1336,6 +1336,19 @@ func (re *RulesEngine) evaluateOperation(
 	operation any,
 	ruleCtx *contexter.RuleContext,
 ) (any, error) {
+	// Add panic recovery to prevent the whole evaluation from failing
+	defer func() {
+		if r := recover(); r != nil {
+			re.logger.WithIndent().Errorf(ctx, "Recovered from panic in evaluateOperation: %v", r)
+		}
+	}()
```

Comment:
Ik ben zelf geen fan van panics omdat je dan undefined states kan krijgen in mutexes etc doordat defers niet goed worden uit gevoerd. 

Is dit nog nodig of was dit voor testen?